Framework/DataHandling/src/PatchBBY.cpp

```diff
@@ -154,22 +188,40 @@ void PatchBBY::exec() {
   std::ostringstream logContentNewBuffer;
   int tmp_int;
   double tmp_dbl;
+  std::string tmp_str;
   for (auto itr = std::begin(PatchableProperties);
        itr != std::end(PatchableProperties); ++itr) {
     auto property_value = getProperty(itr->Name);
+
     // if (!isEmpty(property_value))
     switch (itr->Type) {
     case TYPE_INT:
       tmp_int = property_value;
-      if (tmp_int != EMPTY_INT()) // !!!
+      if (tmp_int != EMPTY_INT())
         logContentNewBuffer << itr->Name << " = " << tmp_int << std::endl;
       break;
 
     case TYPE_DBL:
       tmp_dbl = property_value;
-      if (tmp_dbl != EMPTY_DBL()) // !!!
+      if (tmp_dbl != EMPTY_DBL())
         logContentNewBuffer << itr->Name << " = " << tmp_dbl << std::endl;
       break;
+
+    case TYPE_STR:
+      if (std::strcmp(itr->Name, "FrameSource") != 0)
+        throw std::invalid_argument(std::string("not implemented: ") +
+                                    itr->Name);
+
+      tmp_str = static_cast<std::string>(property_value);
+      if (!tmp_str.empty()) {
+        if (Poco::icompare(tmp_str, EXTERNAL) == 0)
+          logContentNewBuffer << itr->Name << " = " << EXTERNAL << std::endl;
+        else if (Poco::icompare(tmp_str, INTERNAL) == 0)
+          logContentNewBuffer << itr->Name << " = " << INTERNAL << std::endl;
+        else
+          throw std::invalid_argument("invalid: FrameSource");
+      }
+      break;
```

Comment:
the branching in this for-loop is a lot. It might be worth to refactor it when this code is touched again.